File: server/src/main/java/com/content/security/entity/Distributionproduct.java
package com.content.security.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.LocalDate;

@Getter
@Setter
@Entity
@Table(name = "distributionproduct")
public class Distributionproduct {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @JsonIgnore
    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "distribution_id", nullable = false)
    private Distribution distribution;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @Column(name = "date")
    private LocalDate date;

    @Column(name = "quentity")
    private Integer quentity;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "distributionproductstatus_id", nullable = false)
    private Distributionproductstatus distributionproductstatus;

}

File: client/src/app/core/entity/distributionproduct.ts
import {Product} from "./product";
import {DistributionProductStatus} from "./distributionproductstatus";

export class DistributionProduct{

  id: number;
  product:Product;
  quentity:number;
  date:string;
  distributionproductstatus:DistributionProductStatus;

  constructor(id:number,product:Product,quentity:number,date:string,distributionproductstatus:DistributionProductStatus) {
    this.id = id;
    this.product = product;
    this.quentity = quentity;
    this.date = date;
    this.distributionproductstatus = distributionproductstatus;
  }

}


